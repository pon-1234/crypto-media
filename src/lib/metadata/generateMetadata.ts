import { Metadata } from 'next'

/**
 * OGP画像生成のパラメータ
 */
interface OgImageParams {
  title: string
  description?: string
  type?: 'default' | 'article' | 'corporate' | 'media'
  category?: string
}

/**
 * URLSearchParamsを生成するヘルパー関数
 */
function createOgImageUrl(params: OgImageParams): string {
  const baseUrl = process.env.NEXT_PUBLIC_BASE_URL || 'https://crypto-media.jp'
  const searchParams = new URLSearchParams()

  searchParams.append('title', params.title)
  if (params.description) searchParams.append('description', params.description)
  if (params.type) searchParams.append('type', params.type)
  if (params.category) searchParams.append('category', params.category)

  return `${baseUrl}/api/og?${searchParams.toString()}`
}

/**
 * ページメタデータ生成のオプション
 */
export interface GenerateMetadataOptions {
  title: string
  description: string
  path?: string
  ogType?: 'website' | 'article'
  ogImageParams?: OgImageParams
  ogImage?: string
  keywords?: string[]
  noindex?: boolean
  publishedTime?: string
  modifiedTime?: string
  authors?: string[]
  category?: string
  tags?: string[]
}

/**
 * 統一されたメタデータを生成するヘルパー関数
 * @doc https://nextjs.org/docs/app/api-reference/functions/generate-metadata
 * @related src/app/api/og/route.tsx
 */
export function generatePageMetadata(
  options: GenerateMetadataOptions
): Metadata {
  const baseUrl = process.env.NEXT_PUBLIC_BASE_URL || 'https://crypto-media.jp'
  const {
    title,
    description,
    path = '',
    ogType = 'website',
    ogImageParams,
    ogImage,
    keywords = [],
    noindex = false,
    publishedTime,
    modifiedTime,
    authors = [],
    category,
    tags = [],
  } = options

  const url = `${baseUrl}${path}`

  // OGP画像のURL生成
  const ogImageUrl = ogImage
    ? ogImage
    : ogImageParams
    ? createOgImageUrl(ogImageParams)
    : createOgImageUrl({ title, description, type: 'default' })

  const metadata: Metadata = {
    title,
    description,
    keywords: keywords.join(', '),
    openGraph: {
      title,
      description,
      url,
      siteName: 'Crypto Media',
      locale: 'ja_JP',
      type: ogType,
      images: [
        {
          url: ogImageUrl,
          width: 1200,
          height: 630,
          alt: title,
        },
      ],
    },
    twitter: {
      card: 'summary_large_image',
      title,
      description,
      images: [ogImageUrl],
      site: '@cryptomedia_jp',
      creator: '@cryptomedia_jp',
    },
    alternates: {
      canonical: url,
    },
    robots: {
      index: !noindex,
      follow: !noindex,
      googleBot: {
        index: !noindex,
        follow: !noindex,
        'max-video-preview': -1,
        'max-image-preview': 'large',
        'max-snippet': -1,
      },
    },
  }

  // 記事固有のメタデータ
  if (ogType === 'article' && metadata.openGraph) {
    metadata.openGraph = {
      ...metadata.openGraph,
      type: 'article',
      publishedTime,
      modifiedTime,
      authors,
      tags,
    }

    if (category) {
      metadata.openGraph.section = category
    }
  }

  return metadata
}
