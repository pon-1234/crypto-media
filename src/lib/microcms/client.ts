/**
 * microCMS クライアント設定
 * @doc DEVELOPMENT_GUIDE.md#microCMS
 * @issue #2 - microCMSクライアントと型定義の実装
 */
import { createClient } from 'microcms-js-sdk'

if (!process.env.MICROCMS_SERVICE_DOMAIN) {
  throw new Error('MICROCMS_SERVICE_DOMAIN is required')
}

if (!process.env.MICROCMS_API_KEY) {
  throw new Error('MICROCMS_API_KEY is required')
}

/**
 * microCMS クライアントインスタンス
 * @see https://github.com/microcmsio/microcms-js-sdk
 */
export const client = createClient({
  serviceDomain: process.env.MICROCMS_SERVICE_DOMAIN,
  apiKey: process.env.MICROCMS_API_KEY,
})

/**
 * プレビューモードでの記事取得について
 * 
 * プレビューモードでは通常のクライアントを使用し、
 * APIメソッド呼び出し時にdraftKeyをクエリパラメータとして渡します。
 * 
 * 例:
 * ```typescript
 * const preview = await client.get({
 *   endpoint: 'media_articles',
 *   contentId: 'xxx',
 *   queries: { draftKey: 'yyy' }
 * })
 * ```
 */

/**
 * API呼び出しの共通オプション
 */
export const defaultQueries = {
  limit: 100,
  orders: '-publishedAt', // 最新順
} as const

/**
 * リスト取得時の最大件数
 */
export const MAX_LIMIT = 100

/**
 * 画像最適化用のパラメータを生成
 * @param url - microCMSの画像URL
 * @param options - 画像変換オプション
 */
export const getOptimizedImageUrl = (
  url: string,
  options: {
    width?: number
    height?: number
    format?: 'webp' | 'jpg' | 'png'
    quality?: number
  } = {}
): string => {
  const params = new URLSearchParams()
  
  if (options.width) params.append('w', options.width.toString())
  if (options.height) params.append('h', options.height.toString())
  if (options.format) params.append('fm', options.format)
  if (options.quality) params.append('q', options.quality.toString())
  
  return `${url}?${params.toString()}`
}