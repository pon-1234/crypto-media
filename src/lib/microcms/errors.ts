/**
 * @doc microCMS error handling utilities
 * @issue Gemini review recommendation - Better error handling
 */

/**
 * microCMS API error class
 */
export class MicroCMSApiError extends Error {
  constructor(
    message: string,
    public status: number,
    public code?: string
  ) {
    super(message)
    this.name = 'MicroCMSApiError'
  }
}

/**
 * Check if error is a microCMS 404 error
 */
export function isMicroCMS404Error(error: unknown): boolean {
  if (error instanceof MicroCMSApiError) {
    return error.status === 404
  }
  
  // Fallback for microCMS SDK errors
  if (error instanceof Error) {
    // Check for common 404 patterns in error messages
    const message = error.message.toLowerCase()
    return (
      message.includes('404') ||
      message.includes('not found') ||
      message.includes('content not found')
    )
  }
  
  return false
}

/**
 * Check if error is a microCMS API error
 */
export function isMicroCMSError(error: unknown): error is MicroCMSApiError {
  return error instanceof MicroCMSApiError
}

/**
 * Parse microCMS error response
 */
export function parseMicroCMSError(error: unknown): MicroCMSApiError {
  // If it's already a MicroCMSApiError, return it as-is
  if (error instanceof MicroCMSApiError) {
    return error
  }
  
  // Handle Axios-like errors with response property
  if (
    error &&
    typeof error === 'object' &&
    'response' in error &&
    error.response &&
    typeof error.response === 'object' &&
    'status' in error.response
  ) {
    const response = error.response as { status: number; data?: { message?: string } }
    return new MicroCMSApiError(
      response.data?.message || 'API Error',
      response.status
    )
  }

  // Handle standard Error objects
  if (error instanceof Error) {
    // Try to extract status from error message
    const statusMatch = error.message.match(/(\d{3})/)
    const status = statusMatch ? parseInt(statusMatch[1], 10) : 500
    
    return new MicroCMSApiError(error.message, status)
  }

  // Fallback for unknown errors
  return new MicroCMSApiError('Unknown error occurred', 500)
}