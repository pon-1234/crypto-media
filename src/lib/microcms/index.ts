/**
 * microCMS ライブラリのエントリーポイント
 * @doc DEVELOPMENT_GUIDE.md#microCMS
 * @issue #2 - microCMSクライアントと型定義の実装
 */

// クライアント
export { client, defaultQueries, MAX_LIMIT, getOptimizedImageUrl } from './client'

// スキーマとバリデーション
export {
  // スキーマ
  corporateNewsSchema,
  expertSchema,
  categorySchema,
  tagSchema,
  featureSchema,
  mediaArticleSchema,
  siteSettingsSchema,
  researchSurveySchema,
  pageCorporateSchema,
  
  // リストレスポンススキーマ
  corporateNewsListSchema,
  expertListSchema,
  categoryListSchema,
  tagListSchema,
  featureListSchema,
  mediaArticleListSchema,
  researchSurveyListSchema,
  pageCorporateListSchema,
  
  // ヘルパー関数
  createListResponseSchema,
  
  // 型
  type CorporateNews,
  type Expert,
  type Category,
  type Tag,
  type Feature,
  type MediaArticle,
  type SiteSettings,
  type ResearchSurvey,
  type PageCorporate,
} from '@/lib/schema'

// 追加の型エクスポート
export type {
  MicroCMSBase,
  MicroCMSListResponse,
  MicroCMSImage,
  ExpertRole,
  ArticleType,
  MembershipLevel,
} from '@/lib/schema'

// APIメソッド
export {
  getCorporateNewsList,
  getCorporateNewsDetail,
  getAllCorporateNewsIds,
} from './corporate-news'