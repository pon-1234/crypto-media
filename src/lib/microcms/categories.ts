import { client } from './client'
import type { Category } from '@/lib/schema/category.schema'
import type {
  MicroCMSQueries,
  MicroCMSListResponse,
  MicroCMSContentId,
} from 'microcms-js-sdk'

/**
 * Get all categories
 *
 * @doc Fetches all available categories from microCMS
 * @related src/lib/schema/category.schema.ts - Category type definition
 * @param queries - Optional microCMS query parameters
 * @returns Promise containing list of categories
 */
export const getCategories = async (
  queries?: MicroCMSQueries
): Promise<MicroCMSListResponse<Category>> => {
  return await client.getList<Category>({
    endpoint: 'categories',
    queries,
  })
}

/**
 * Get a single category by slug
 *
 * @doc Fetches a specific category by its slug identifier
 * @related src/lib/schema/category.schema.ts - Category type definition
 * @param slug - The category slug to fetch
 * @returns Promise containing the category data or null if not found
 */
export const getCategoryBySlug = async (
  slug: string
): Promise<Category | null> => {
  try {
    const response = await client.getList<Category>({
      endpoint: 'categories',
      queries: {
        filters: `slug[equals]${slug}`,
        limit: 1,
      },
    })
    return response.contents[0] || null
  } catch (error) {
    console.error('Error fetching category by slug:', error)
    return null
  }
}

/**
 * Get a single category by ID
 *
 * @doc Fetches a specific category by its microCMS ID
 * @related src/lib/schema/category.schema.ts - Category type definition
 * @param id - The microCMS content ID
 * @param queries - Optional microCMS query parameters
 * @returns Promise containing the category data
 */
export const getCategoryById = async (
  id: string,
  queries?: MicroCMSQueries
): Promise<Category & MicroCMSContentId> => {
  return await client.get<Category>({
    endpoint: 'categories',
    contentId: id,
    queries,
  })
}