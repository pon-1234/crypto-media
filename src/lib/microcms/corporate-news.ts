import {
  corporateNewsListSchema,
  corporateNewsSchema,
  type CorporateNews,
} from '@/lib/schema'
import type { MicroCMSQueries } from 'microcms-js-sdk'
import { client } from './client'
import { handleError } from '@/lib/utils/handleError'

/**
 * コーポレートお知らせ一覧を取得する
 *
 * @issue #4 - コーポレートお知らせ一覧・詳細ページの実装
 */
export async function getCorporateNewsList(queries?: MicroCMSQueries): Promise<{
  contents: CorporateNews[]
  totalCount: number
  offset: number
  limit: number
}> {
  try {
    const response = await client.get({
      endpoint: 'corporate_news',
      queries,
    })

    return corporateNewsListSchema.parse(response)
  } catch (error) {
    handleError(error, 'Failed to fetch corporate news list')
    throw error
  }
}

/**
 * コーポレートお知らせの詳細を取得する
 *
 * @issue #4 - コーポレートお知らせ一覧・詳細ページの実装
 */
export async function getCorporateNewsDetail(
  contentId: string,
  queries?: MicroCMSQueries
): Promise<CorporateNews> {
  try {
    const response = await client.get({
      endpoint: 'corporate_news',
      contentId,
      queries,
    })

    return corporateNewsSchema.parse(response)
  } catch (error) {
    handleError(
      error,
      `Failed to fetch corporate news detail for id: ${contentId}`
    )
    throw error
  }
}

/**
 * すべてのコーポレートお知らせのIDを取得する
 * generateStaticParams用
 *
 * @issue #4 - コーポレートお知らせ一覧・詳細ページの実装
 */
export async function getAllCorporateNewsIds(): Promise<string[]> {
  try {
    const response = await client.get({
      endpoint: 'corporate_news',
      queries: {
        fields: 'id',
        limit: 100, // 必要に応じて調整
      },
    })

    // fields: 'id'を指定した場合は、idのみが返るためスキーマ検証をスキップ
    return response.contents.map((item: { id: string }) => item.id)
  } catch (error) {
    handleError(error, 'Failed to fetch corporate news IDs')
    throw error
  }
}

/**
 * カテゴリ別にコーポレートお知らせ一覧を取得する
 * TODO: microCMSにcategoryフィールドを追加後に実装を有効化
 *
 * @issue #35 - コーポレートサイトの未実装ページ作成
 */
export async function getCorporateNewsListByCategory(
  category: string,
  queries?: MicroCMSQueries
): Promise<{
  contents: CorporateNews[]
  totalCount: number
  offset: number
  limit: number
}> {
  try {
    // TODO: categoryフィールド追加後に以下のコメントを解除
    // const response = await client.get({
    //   endpoint: 'corporate_news',
    //   queries: {
    //     ...queries,
    //     filters: `category[equals]${category}`,
    //   },
    // })

    // 仮実装: 通常の一覧を返す
    const response = await client.get({
      endpoint: 'corporate_news',
      queries,
    })

    return corporateNewsListSchema.parse(response)
  } catch (error) {
    handleError(
      error,
      `Failed to fetch corporate news list for category: ${category}`
    )
    throw error
  }
}
