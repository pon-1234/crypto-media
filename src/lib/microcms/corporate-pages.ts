/**
 * @doc Corporate pages API client methods
 * @related DEVELOPMENT_GUIDE.md#pages_corporate
 * @issue https://github.com/pon-1234/crypto-media/issues/25
 */
import { client, defaultQueries } from './client'
import {
  corporatePageSchema,
  corporatePageListSchema,
  type CorporatePage,
  type CorporatePageList,
} from '@/lib/schema/corporate-page.schema'
import { isMicroCMS404Error, parseMicroCMSError } from './errors'

/**
 * Corporate page API query parameters
 */
type CorporatePageQueries = {
  draftKey?: string
  limit?: number
  offset?: number
}

/**
 * Get all corporate pages
 */
export async function getCorporatePages(
  queries?: CorporatePageQueries
): Promise<CorporatePageList> {
  const response = await client.get({
    endpoint: 'pages_corporate',
    queries: {
      ...defaultQueries,
      ...queries,
    },
  })

  return corporatePageListSchema.parse(response)
}

/**
 * Get a single corporate page by slug
 */
export async function getCorporatePageBySlug(
  slug: string,
  queries?: Pick<CorporatePageQueries, 'draftKey'>
): Promise<CorporatePage | null> {
  try {
    const response = await client.get({
      endpoint: 'pages_corporate',
      queries: {
        filters: `slug[equals]${slug}`,
        limit: 1,
        ...queries,
      },
    })

    const parsed = corporatePageListSchema.parse(response)
    return parsed.contents[0] || null
  } catch (error) {
    const microCMSError = parseMicroCMSError(error)
    if (isMicroCMS404Error(microCMSError)) {
      return null
    }
    throw microCMSError
  }
}

/**
 * Get a single corporate page by ID (for preview mode)
 */
export async function getCorporatePageById(
  id: string,
  queries?: Pick<CorporatePageQueries, 'draftKey'>
): Promise<CorporatePage | null> {
  try {
    const response = await client.get({
      endpoint: 'pages_corporate',
      contentId: id,
      queries,
    })

    return corporatePageSchema.parse(response)
  } catch (error) {
    const microCMSError = parseMicroCMSError(error)
    if (isMicroCMS404Error(microCMSError)) {
      return null
    }
    throw microCMSError
  }
}

/**
 * Check if a corporate page exists by slug
 */
export async function corporatePageExists(slug: string): Promise<boolean> {
  const page = await getCorporatePageBySlug(slug)
  return page !== null
}
