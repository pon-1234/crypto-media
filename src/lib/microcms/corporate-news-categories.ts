/**
 * コーポレートニュースカテゴリ用のAPIクライアント
 * @doc コーポレートニュースのカテゴリ管理機能
 * @related corporate-news.ts
 * @issue #41 - コーポレートニュースのカテゴリ機能
 */
import { client } from './client'
import { corporateNewsCategoryListSchema } from '@/lib/schema'
import { getAllContents } from './utils'
import type { CorporateNewsCategory } from '@/lib/schema'

/**
 * コーポレートニュースカテゴリをスラッグで取得
 * @doc 指定されたスラッグに一致するカテゴリを取得
 * @param slug カテゴリのスラッグ
 * @returns カテゴリオブジェクト、存在しない場合はnull
 */
export async function getCorporateNewsCategoryBySlug(
  slug: string
): Promise<CorporateNewsCategory | null> {
  const response = await client.getList({
    endpoint: 'corporate_news_categories',
    queries: { filters: `slug[equals]${slug}`, limit: 1 },
  })
  const validated = corporateNewsCategoryListSchema.parse(response)
  return validated.contents[0] || null
}

/**
 * すべてのカテゴリのスラッグを取得
 * @doc 静的生成のためのスラッグ一覧取得
 * @returns スラッグの配列
 */
export async function getAllCorporateNewsCategorySlugs(): Promise<string[]> {
  const allCategories = await getAllContents<{ id: string; slug: string }>(
    'corporate_news_categories',
    {
      fields: 'id,slug',
    }
  )
  return allCategories.map((c) => c.slug)
}