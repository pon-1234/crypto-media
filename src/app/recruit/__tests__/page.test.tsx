/**
 * @vitest-environment jsdom
 */
import { describe, it, expect, vi } from 'vitest'
import { notFound } from 'next/navigation'
import RecruitPage, { generateMetadata } from '../page'
import { getCorporatePageBySlug } from '@/lib/microcms/corporate-pages'
import { generatePageMetadata } from '@/lib/metadata/generateMetadata'
import type { CorporatePage } from '@/lib/schema/corporate-page.schema'

vi.mock('next/navigation', () => ({
  notFound: vi.fn(() => {
    throw new Error('NEXT_NOT_FOUND')
  }),
}))

vi.mock('@/lib/microcms/corporate-pages', () => ({
  getCorporatePageBySlug: vi.fn(),
}))

vi.mock('@/lib/metadata/generateMetadata', () => ({
  generatePageMetadata: vi.fn(),
}))

vi.mock('@/components/corporate/CorporatePageContent', () => ({
  CorporatePageContent: vi.fn(({ page }) => (
    <div data-testid="corporate-page-content">{page.content}</div>
  )),
}))

describe('RecruitPage', () => {
  const mockPage: CorporatePage = {
    id: '1',
    slug: 'recruit',
    title: '採用情報',
    description: '当社の採用情報をご紹介します',
    content: '<p>採用情報の内容</p>',
    createdAt: '2024-01-01T00:00:00Z',
    updatedAt: '2024-01-01T00:00:00Z',
    publishedAt: '2024-01-01T00:00:00Z',
    revisedAt: '2024-01-01T00:00:00Z',
    metadata: {
      ogImage: { url: 'https://example.com/og-image.jpg', width: 1200, height: 630 },
      keywords: ['採用', 'キャリア'],
    },
  }

  it('採用情報ページが正しく表示される', async () => {
    vi.mocked(getCorporatePageBySlug).mockResolvedValue(mockPage)

    const page = await RecruitPage()

    expect(getCorporatePageBySlug).toHaveBeenCalledWith('recruit')
    expect(page).toMatchSnapshot()
  })

  it('ページが見つからない場合はnotFoundを呼ぶ', async () => {
    vi.mocked(getCorporatePageBySlug).mockResolvedValue(null)

    await expect(RecruitPage()).rejects.toThrow('NEXT_NOT_FOUND')
    expect(notFound).toHaveBeenCalled()
  })

  describe('generateMetadata', () => {
    it('ページが存在する場合、適切なメタデータを生成する', async () => {
      vi.mocked(getCorporatePageBySlug).mockResolvedValue(mockPage)
      const mockMetadata = {
        title: '採用情報 | 株式会社Example',
        description: '当社の採用情報をご紹介します',
      }
      vi.mocked(generatePageMetadata).mockReturnValue(mockMetadata)

      const metadata = await generateMetadata()

      expect(getCorporatePageBySlug).toHaveBeenCalledWith('recruit')
      expect(generatePageMetadata).toHaveBeenCalledWith({
        title: '採用情報',
        description: '当社の採用情報をご紹介します',
        path: '/recruit',
        ogImage: 'https://example.com/og-image.jpg',
        keywords: ['採用', 'キャリア'],
      })
      expect(metadata).toEqual(mockMetadata)
    })

    it('ページが見つからない場合は空のメタデータを返す', async () => {
      vi.clearAllMocks() // モックをクリア
      vi.mocked(getCorporatePageBySlug).mockResolvedValue(null)

      const metadata = await generateMetadata()

      expect(metadata).toEqual({})
      expect(generatePageMetadata).not.toHaveBeenCalled()
    })

    it('descriptionがない場合は空文字を渡す', async () => {
      const pageWithoutDescription = { ...mockPage, description: undefined }
      vi.mocked(getCorporatePageBySlug).mockResolvedValue(pageWithoutDescription)

      await generateMetadata()

      expect(generatePageMetadata).toHaveBeenCalledWith(
        expect.objectContaining({
          description: '',
        })
      )
    })

    it('メタデータがない場合はundefinedを渡す', async () => {
      const pageWithoutMetadata = { ...mockPage, metadata: undefined }
      vi.mocked(getCorporatePageBySlug).mockResolvedValue(pageWithoutMetadata)

      await generateMetadata()

      expect(generatePageMetadata).toHaveBeenCalledWith(
        expect.objectContaining({
          ogImage: undefined,
          keywords: undefined,
        })
      )
    })
  })
})