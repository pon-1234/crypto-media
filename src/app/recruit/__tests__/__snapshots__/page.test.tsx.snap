// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RecruitPage > 採用情報ページが正しく表示される 1`] = `
<main
  className="container mx-auto px-4 py-8"
>
  <div
    className="mx-auto max-w-4xl"
  >
    <h1
      className="mb-8 text-3xl font-bold"
    >
      採用情報
    </h1>
    <spy
      page={
        {
          "content": "<p>採用情報の内容</p>",
          "createdAt": "2024-01-01T00:00:00Z",
          "description": "当社の採用情報をご紹介します",
          "id": "1",
          "metadata": {
            "keywords": [
              "採用",
              "キャリア",
            ],
            "ogImage": {
              "height": 630,
              "url": "https://example.com/og-image.jpg",
              "width": 1200,
            },
          },
          "publishedAt": "2024-01-01T00:00:00Z",
          "revisedAt": "2024-01-01T00:00:00Z",
          "slug": "recruit",
          "title": "採用情報",
          "updatedAt": "2024-01-01T00:00:00Z",
        }
      }
    />
  </div>
</main>
`;
